src/System.Net.Http/src/System/Net/Http/Managed/AuthenticationHandler.cs

```diff
@@ -53,17 +49,26 @@ private bool TrySetBasicAuthToken(HttpRequestMessage request)
 
                 foreach (AuthenticationHeaderValue h in authenticateValues)
                 {
-                    // We only support Basic auth, ignore others
-                    if (h.Scheme == Basic)
+                    // We only support Basic and digest auth, ignore others
+                    if (h.Scheme == AuthenticationHelper.Basic)
                     {
-                        if (!TrySetBasicAuthToken(request))
+                        if (AuthenticationHelper.TrySetBasicAuthToken(request, _credentials))
                         {
+                            response.Dispose();
```

Comment:
It's worthwhile if using SSL because of the extra handshake. Also if we implement NTLM authentication, then you authenticate the socket itself so you shouldn't close the socket otherwise you restart auth.